src/main/java/org/cryptomator/cryptofs/health/dirid/OrphanDir.java

```diff
@@ -108,45 +110,47 @@ public void fix(Path pathToVault, VaultConfig config, Masterkey masterkey, Crypt
 		Path cipherRecoveryDir = dataDir.resolve(recoveryDirHash.substring(0, 2)).resolve(recoveryDirHash.substring(2)).toAbsolutePath();
 		Files.createDirectories(cipherRecoveryDir);
 
+		return cipherRecoveryDir;
+	}
+
+	// visible for testing
+	CryptoPathMapper.CiphertextDirectory prepareStepParent(Path cipherRecoveryDir, FileNameCryptor cryptor, String clearStepParentDirName) throws IOException {
 		//create "step-parent" directory to move orphaned files to
 		String cipherStepParentDirName = convertClearToCiphertext(cryptor, clearStepParentDirName, Constants.RECOVERY_DIR_ID);
 		Path cipherStepParentDirFile = cipherRecoveryDir.resolve(cipherStepParentDirName + "/" + Constants.DIR_FILE_NAME);
 		final String stepParentUUID;
 		if (Files.exists(cipherStepParentDirFile, LinkOption.NOFOLLOW_LINKS)) {
 			stepParentUUID = Files.readString(cipherStepParentDirFile, StandardCharsets.UTF_8);
 		} else {
-			Files.createDirectory(cipherStepParentDirFile.getParent());
+			Files.createDirectories(cipherStepParentDirFile.getParent());
 			stepParentUUID = UUID.randomUUID().toString();
 			Files.writeString(cipherStepParentDirFile, stepParentUUID, StandardCharsets.UTF_8, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE);
 		}
 		String stepParentDirHash = cryptor.hashDirectoryId(stepParentUUID);
-		Path stepParentDir = dataDir.resolve(stepParentDirHash.substring(0, 2)).resolve(stepParentDirHash.substring(2)).toAbsolutePath();
+		Path stepParentDir = cipherRecoveryDir.getParent().getParent().resolve(stepParentDirHash.substring(0, 2)).resolve(stepParentDirHash.substring(2)).toAbsolutePath();
 		Files.createDirectories(stepParentDir);
 		return new CryptoPathMapper.CiphertextDirectory(stepParentUUID, stepParentDir);
 	}
 
 	// visible for testing
-	void adoptOrphanedResource(Adoption adoption, CryptoPathMapper.CiphertextDirectory stepParentDir, FileNameCryptor cryptor, String longNameSuffix, MessageDigest sha1) throws IOException {
-		if (adoption.oldCipherPath.toString().endsWith(Constants.DEFLATED_FILE_SUFFIX)) {
-			var newCipherName = convertClearToCiphertext(cryptor, adoption.newClearname + longNameSuffix, stepParentDir.dirId);
+	void adoptOrphanedResource(Path oldCipherPath, String newClearname, CryptoPathMapper.CiphertextDirectory stepParentDir, FileNameCryptor cryptor, String longNameSuffix, MessageDigest sha1) throws IOException {
+		if (oldCipherPath.toString().endsWith(Constants.DEFLATED_FILE_SUFFIX)) {
+			var newCipherName = convertClearToCiphertext(cryptor, newClearname + longNameSuffix, stepParentDir.dirId);
 			var deflatedName = BaseEncoding.base64Url().encode(sha1.digest(newCipherName.getBytes(StandardCharsets.UTF_8))) + Constants.DEFLATED_FILE_SUFFIX;
 			Path targetPath = stepParentDir.path.resolve(deflatedName);
-			Files.move(adoption.oldCipherPath, targetPath, StandardCopyOption.ATOMIC_MOVE);
+			Files.move(oldCipherPath, targetPath, StandardCopyOption.ATOMIC_MOVE);
```

Comment:
I believe ATOMIC_MOVE may thrown an exception, if the fs doesn't support it.

Reply:
What do you suggest? I vote for a retry without the parameter, maybe even caching the setting afterwards.

Reply:
I'm not entirely sure about my assumption. But if I'm not mistaken, I'd vote for just going straight for non-atomic moves. Since we move files to targets that we know for sure don't exist yet and furthermore will not access these files until the user exists health check and unlocks the vault again, there shouldn't be much need to avoid race conditions. Treat the file system as eventually consistent.

Reply:
Good point about the access restriction. I'll remove the option.